std/typecons.d

```diff
@@ -3703,6 +3815,69 @@ unittest
     assert(i.bar(10) == 100);
 }
 
+// Concat all Targets function members into one tuple
+private template ConcatInterfaceMembers(Targets...)
+{
+    static if (Targets.length == 0)
+        alias ConcatInterfaceMembers = TypeTuple!();
+    else static if (Targets.length == 1)
+        alias ConcatInterfaceMembers
+          = TypeTuple!(GetOverloadedMethods!(Targets[0]));
+    else
+        alias ConcatInterfaceMembers = TypeTuple!(
+                GetOverloadedMethods!(Targets[0]),
+                ConcatInterfaceMembers!(Targets[1..$]));
+}
+// Remove duplicated functions based on the identifier name and function type covariance
+private template UniqMembers(members...)
+{
+            template FuncInfo(string s, F)
+            {
+                enum name = s;
+                alias type = F;
+            }
+
+    static if (members.length == 0)
+        alias UniqMembers = TypeTuple!();
+    else
+    {
+        alias func = members[0];
+        enum  name = __traits(identifier, func);
+        alias type = FunctionTypeOf!func;
+        template check(size_t i, mem...)
+        {
+            static if (i >= mem.length)
+                enum ptrdiff_t check = -1;
+            else
+            {
+                enum ptrdiff_t check =
```

Comment:
With this complex a condition, it might be worth considering an inner static if statement for clarity.